client/state/reader/posts/normalization-rules.js

```diff
@@ -96,7 +97,7 @@ export function classifyPost( post ) {
 		displayType ^= DISPLAY_TYPES.FEATURED_VIDEO;
 	}
 
-	if ( post.content_images && filter( post.content_images, imageIsBigEnoughForGallery ).length >= GALLERY_MIN_IMAGES ) {
+	if ( post.content_images && imagesForGallery.length >= GALLERY_MIN_IMAGES ) {
```

Comment:
idea to simplify this section:

```javascript

if ( imagesForGallery.length >= GALLERY_MIN_IMAGES ) { // drop the post.content_images truthy check
 ...
} else if ( ... PHOTO_ONLY stuff) { // we can else-if them because it should never be both

}

```